tests/lti/models/test_oauth2_unvalidated_credentials.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from lti.models import OAuth2UnvalidatedCredentials
+
+
+class TestOAuthUnvalidated2Credentials(object):
+
+    def test_it_persists_the_fields(self, db_session):
+        db_session.add(OAuth2UnvalidatedCredentials(client_id="TEST_ID",
+                                                    client_secret="TEST_SECRET",
+                                                    authorization_server="TEST_AUTH_SERVER",
+                                                    email_address="TEST_EMAIL",
+        ))
+
+        persisted = (db_session.query(OAuth2UnvalidatedCredentials)
+                               .filter_by(client_id="TEST_ID").one())
```

Comment:
@sheetaluk And in the tests `request.db` is called `db_session`. Maybe I should make it available as `pyramid_request.db` in the tests as well.

Also you can see here how you read a saved object out of the database: you do `db_session.query(SomeModelClass)` or `request.db.query(SomeModelClass)` and then you add these query filter methods on like `.one()` or `.all()` etc.

You might find this tutorial helpful for figuring out what's going on here: <https://docs.sqlalchemy.org/en/rel_1_1/orm/tutorial.html>

I'll be able to explain more when I've finished the DB work on the `AuthData` class as well